packages/ui/src/components/sign-up/sign-up.tsx

```diff
@@ -1,34 +1,51 @@
 import { useClerk } from '@clerk/clerk-react';
 import * as Common from '@clerk/elements/common';
 import * as SignUp from '@clerk/elements/sign-up';
+import type { EnvironmentResource } from '@clerk/types';
 
 import { EmailField } from '~/common/email-field';
 import { FirstNameField } from '~/common/first-name-field';
 import { LastNameField } from '~/common/last-name-field';
 import { OTPField } from '~/common/otp-field';
 import { PasswordField } from '~/common/password-field';
+import { PhoneNumberField } from '~/common/phone-number-field';
+import { UsernameField } from '~/common/username-field';
 import { PROVIDERS } from '~/constants';
-import { getEnabledSocialConnectionsFromEnvironment } from '~/hooks/getEnabledSocialConnectionsFromEnvironment';
 import { Button } from '~/primitives/button';
 import * as Card from '~/primitives/card';
 import * as Connection from '~/primitives/connection';
 import * as Icon from '~/primitives/icon';
 import { Seperator } from '~/primitives/seperator';
+import { TextButton } from '~/primitives/text-button';
+import { getEnabledSocialConnectionsFromEnvironment } from '~/utils/getEnabledSocialConnectionsFromEnvironment';
 
 export function SignUpComponent() {
-  const clerk = useClerk();
-  // @ts-ignore clerk is not null
-  const enabledConnections = getEnabledSocialConnectionsFromEnvironment(clerk?.__unstable__environment);
-
   return (
     <SignUp.Root>
-      <Common.Loading>
-        {isGlobalLoading => {
-          return (
-            <SignUp.Step
-              data-clerk
-              name='start'
-            >
+      <SignUpComponentLoaded />
+    </SignUp.Root>
+  );
+}
+
+function SignUpComponentLoaded() {
+  const clerk = useClerk();
+  const enabledConnections = getEnabledSocialConnectionsFromEnvironment(
+    (clerk as any)?.__unstable__environment as EnvironmentResource,
+  );
+  const locationBasedCountryIso = (clerk as any)?.__internal_country;
+  const attributes = ((clerk as any)?.__unstable__environment as EnvironmentResource)?.userSettings.attributes;
+  const { enabled: firstNameEnabled, required: firstNameRequired } = attributes['first_name'];
+  const { enabled: lastNameEnabled, required: lastNameRequired } = attributes['last_name'];
+  const { enabled: usernameEnabled, required: usernameRequired } = attributes['username'];
+  const { enabled: phoneNumberEnabled, required: phoneNumberRequired } = attributes['phone_number'];
+  const { enabled: emailAddressEnabled, required: emailAddressRequired } = attributes['email_address'];
+  const { enabled: passwordEnabled, required: passwordRequired } = attributes['password'];
```

Comment:
Explored a couple of different components to expose this data, but ultimately decided to keep things simple and just access the data here directly. Theres likely an abstraction that can be made here, but keeping it simple atm.